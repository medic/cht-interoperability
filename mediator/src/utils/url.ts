import url from 'url';

export function generateBasicAuthUrl(endpoint: string, username: string, password: string) {
  if (!username || !password) {
    throw new Error('Invalid username and password');
  }

  const parsedUrl = url.parse(endpoint);
  const urlParts = [];

  const { protocol, hostname, port, path } = parsedUrl;

  urlParts.push(`${protocol}//`);
  urlParts.push(`${username}:${password}@`);
  urlParts.push(`${hostname}`);

  if (port) {
    urlParts.push(`:${port}`);
  }

  urlParts.push(path);

  return urlParts.join('');
}
