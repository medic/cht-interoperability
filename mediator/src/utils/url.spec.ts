import { generateChtRecordsApiUrl } from "./url";

describe("generateChtRecordsApiUrl", () => {
  it("generates a CHT Records API URL when given right chtUrl", () => {

  })
})

describe("generateBasicAuthUrl", () => {
  it("", () => {})
})