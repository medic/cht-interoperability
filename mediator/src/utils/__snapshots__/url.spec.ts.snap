// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateBasicAuthUrl throws an error if the endpoint is undefined 1`] = `"The "url" argument must be of type string. Received undefined"`;

exports[`generateBasicAuthUrl throws an error if the password is invalid or undefined 1`] = `"Invalid username and password"`;

exports[`generateBasicAuthUrl throws an error if the username is invalid or undefined 1`] = `"Invalid username and password"`;
